delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resources/ResourceEventExchange.scala

```diff
@@ -51,10 +52,25 @@ class ResourceEventExchange(resources: Resources)(implicit base: BaseUri) extend
 
   override def toResource(event: Event, tag: Option[TagLabel]): UIO[Option[EventExchangeValue[A, M]]] =
     event match {
-      case ev: ResourceEvent => resourceToValue(resources.fetch(IdSegmentRef.fromTagOpt(ev.id, tag), ev.project, None))
+      case ev: ResourceEvent =>
+        tag match {
+          case None    =>
+            fetch(IdSegmentRef(ev.id), ev.project)
+          case Some(t) =>
+            resources
+              .fetch(IdSegmentRef(ev.id, t), ev.project, None)
+              .map(Resources.eventExchangeValue(_, Some(t)))
+              .redeemWith(
```

Comment:
what's the reason for this `redeemWith`?

Reply:
if the the fetch by tag fails, we fetch the latest version and create an `EventExchangeValue` with None passed as a tag.
It's not very efficient, since if we use a tag in a view, we need to delete from the index for each resource without the given tag.

Reply:
I think we have to change the signature in some way (at least temporarilyy) because in this current version, it is difficult to expect for the consumer to be aware that it is in fact the last version and not the tagged one. We will hit bugs in the long run.

Reply:
`EventExchangeValue` now has `Option[TagLabel]` to signal if the resource is the latest revision or a tagged one. Not sure what would be a better/simpler solution.

Reply:
An `Either` or an ADT somewhere would allow to skip that second call and make things explicit.
Not sure about how to do it without stirring up a hornets' nest in terms of refactoring though :).

Reply:
To skip the second call we probably would have to make `ResourceF` abstract or trait and have `LatestResourceF` and `TaggedResourceF`. That would push a lot of this complexity to `Resources`